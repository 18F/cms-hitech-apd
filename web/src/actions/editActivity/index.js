export * from './contractorResources';
export * from './costAllocate';
export * from './overview';
