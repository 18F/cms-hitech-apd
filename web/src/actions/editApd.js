import { updateBudget } from './apd';

export const ADD_APD_ITEM = Symbol('add apd item');
export const EDIT_APD = Symbol('edit apd');
export const REMOVE_APD_ITEM = Symbol('remove apd item');

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};

export const setMedicaidDirectorEmail = email => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/email',
    value: email
  };
};

export const setMedicaidDirectorName = name => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/name',
    value: name
  };
};

export const setMedicaidDirectorPhoneNumber = phone => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/phone',
    value: phone
  };
};

export const setMedicaidOfficeAddress1 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address1',
    value: address
  };
};

export const setMedicaidOfficeAddress2 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address2',
    value: address
  };
};

export const setMedicaidOfficeCity = city => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/city',
    value: city
  };
};

export const setMedicaidOfficeState = state => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/state',
    value: state
  };
};

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
