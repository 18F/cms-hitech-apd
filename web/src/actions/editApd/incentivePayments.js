import { EDIT_APD } from './symbols';

export const setIncentiveEHCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehCt/${year}/${quarter}`,
  value
});

export const setIncentiveEHPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehAmt/${year}/${quarter}`,
  value
});

export const setIncentiveEPCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epCt/${year}/${quarter}`,
  value
});

export const setIncentiveEPPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epAmt/${year}/${quarter}`,
  value
});
