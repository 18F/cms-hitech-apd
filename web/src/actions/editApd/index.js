export * from './symbols';

export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
