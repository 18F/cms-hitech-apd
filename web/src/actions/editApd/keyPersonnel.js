import { ADD_APD_ITEM, EDIT_APD, REMOVE_APD_ITEM } from './symbols';
import { updateBudget } from '../apd';

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};
