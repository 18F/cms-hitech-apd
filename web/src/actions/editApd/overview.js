import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
import { updateBudget } from '../apd';

export const addYear = year => (dispatch, getState) => {
  dispatch({ type: ADD_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const removeYear = year => (dispatch, getState) => {
  dispatch({ type: REMOVE_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const setNarrativeForHIE = text => ({
  type: EDIT_APD,
  path: '/narrativeHIE',
  value: text
});

export const setNarrativeForHIT = text => ({
  type: EDIT_APD,
  path: '/narrativeHIT',
  value: text
});

export const setNarrativeForMMIS = text => ({
  type: EDIT_APD,
  path: '/narrativeMMIS',
  value: text
});

export const setProgramOverview = text => ({
  type: EDIT_APD,
  path: '/programOverview',
  value: text
});
