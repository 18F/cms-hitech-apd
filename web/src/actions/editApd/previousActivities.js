import { EDIT_APD } from './symbols';

export const setPreviousActivitySummary = summary => ({
  type: EDIT_APD,
  path: '/previousActivitySummary',
  value: summary
});

export const setPreviousActivityApprovedExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/federalActual`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/federalActual`,
  value: expense
});
