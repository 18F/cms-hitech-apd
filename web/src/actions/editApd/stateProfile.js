import { EDIT_APD } from './symbols';

export const setMedicaidDirectorEmail = email => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/email',
  value: email
});

export const setMedicaidDirectorName = name => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/name',
  value: name
});

export const setMedicaidDirectorPhoneNumber = phone => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/phone',
  value: phone
});

export const setMedicaidOfficeAddress1 = address => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/address1',
  value: address
});

export const setMedicaidOfficeAddress2 = address => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/address2',
  value: address
});

export const setMedicaidOfficeCity = city => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/city',
  value: city
});

export const setMedicaidOfficeState = state => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/state',
  value: state
});

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
