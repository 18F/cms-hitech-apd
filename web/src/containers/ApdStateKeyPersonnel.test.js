import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import KeyPersonnel from './ApdStateKeyPersonnel';
import { ADD_APD_ITEM, REMOVE_APD_ITEM } from '../actions/editApd';
import { UPDATE_BUDGET } from '../actions/apd';

const mockStore = configureStore([thunk]);

describe('apd state profile, Medicaid office component', () => {
  const state = {
    apd: {
      data: {
        keyPersonnel: [
          {
            key: 'person1',
            name: 'person name',
            email: 'person1@theplace.gov',
            position: 'unobservable',
            percentTime: 27,
            hasCosts: true,
            costs: { '1': 100, '2': 200 }
          },
          {
            key: 'person2',
            name: '',
            email: 'person2@theplace.gov',
            position: '',
            percentTime: 72,
            hasCosts: false,
            costs: { '1': 0, '2': 0 }
          }
        ],
        years: ['1', '2']
      }
    }
  };

  const store = mockStore(state);

  // Deleting things causes a window.confirm call. Mock that to always return
  // true so we have something to test - we're not interested in testing that
  // the action obeys the confirm call here.
  jest.spyOn(window, 'confirm').mockImplementation(() => true);

  const component = mount(
    <Provider store={store}>
      <KeyPersonnel />
    </Provider>
  );

  beforeEach(() => {
    store.clearActions();
  });

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('events', () => {
    const list = component.find('FormAndReviewList');

    it('handles adding a new key person', () => {
      list.prop('onAddClick')();
      expect(store.getActions()).toEqual([
        {
          type: ADD_APD_ITEM,
          path: '/keyPersonnel/-'
        }
      ]);
    });

    it('handles deleting a key person', () => {
      list.prop('onDeleteClick')('person1');
      expect(store.getActions()).toEqual([
        { type: REMOVE_APD_ITEM, path: '/keyPersonnel/0' },
        { type: UPDATE_BUDGET, state }
      ]);
    });
  });
});
