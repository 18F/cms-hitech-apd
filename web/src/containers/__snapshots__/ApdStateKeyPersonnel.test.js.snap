// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd state profile, Medicaid office component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ApdStateKeyPersonnel>
    <FormAndReviewList
      addButtonText="Add another person"
      allowDeleteAll={false}
      className={null}
      collapsed={[Function]}
      expanded={[Function]}
      list={
        Array [
          Object {
            "costs": Object {
              "1": 100,
              "2": 200,
            },
            "email": "person1@theplace.gov",
            "hasCosts": true,
            "key": "person1",
            "name": "person name",
            "percentTime": 27,
            "position": "unobservable",
          },
          Object {
            "costs": Object {
              "1": 0,
              "2": 0,
            },
            "email": "person2@theplace.gov",
            "hasCosts": false,
            "key": "person2",
            "name": "",
            "percentTime": 72,
            "position": "",
          },
        ]
      }
      noDataMessage={null}
      onAddClick={[Function]}
      onDeleteClick={[Function]}
      years={
        Array [
          "1",
          "2",
        ]
      }
    >
      <div
        className="form-and-review-list "
      >
        <FormAndReviewItem
          collapsedComponent={[Function]}
          expandedComponent={[Function]}
          index={0}
          initialCollapsed={true}
          item={
            Object {
              "costs": Object {
                "1": 100,
                "2": 200,
              },
              "email": "person1@theplace.gov",
              "hasCosts": true,
              "key": "person1",
              "name": "person name",
              "percentTime": 27,
              "position": "unobservable",
            }
          }
          key="person1"
          onDeleteClick={[Function]}
          years={
            Array [
              "1",
              "2",
            ]
          }
        >
          <div
            className="form-and-review-list--item__collapsed"
          >
            <ApdStateKeyPerson
              expand={[Function]}
              index={0}
              item={
                Object {
                  "costs": Object {
                    "1": 100,
                    "2": 200,
                  },
                  "email": "person1@theplace.gov",
                  "hasCosts": true,
                  "key": "person1",
                  "name": "person name",
                  "percentTime": 27,
                  "position": "unobservable",
                }
              }
              onDeleteClick={[Function]}
              years={
                Array [
                  "1",
                  "2",
                ]
              }
            >
              <div
                className="visibility--screen"
              >
                <StandardReview
                  editHref={null}
                  heading="1. person name"
                  headingLevel={4}
                  onDeleteClick={[Function]}
                  onEditClick={[Function]}
                >
                  <Review
                    editContent={
                      <div
                        className="nowrap visibility--screen"
                      >
                        <Button
                          onClick={[Function]}
                          size="small"
                          type="button"
                          variation="transparent"
                        >
                          Edit
                        </Button>
                        <React.Fragment>
                          |
                          <Button
                            onClick={[Function]}
                            size="small"
                            type="button"
                            variation="transparent"
                          >
                            Remove
                          </Button>
                        </React.Fragment>
                      </div>
                    }
                    editText="Edit"
                    heading="1. person name"
                    headingLevel={4}
                  >
                    <div
                      className="ds-c-review ds-u-border-bottom--2 ds-u-padding-y--2 ds-u-justify-content--between ds-u-display--flex"
                    >
                      <div
                        className="ds-u-margin-right--2"
                      >
                        <h4
                          className="ds-c-review__heading ds-text ds-u-margin-bottom--0 ds-u-font-weight--bold ds-u-display--inline-block"
                        >
                          1. person name
                        </h4>
                        <div
                          className="ds-c-review__body"
                        >
                          <ul
                            className="ds-c-list--bare"
                          >
                            <li>
                              Primary APD Point of Contact
                            </li>
                            <li>
                              unobservable
                            </li>
                            <li>
                              <strong>
                                Total cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $300
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $300
                                </span>
                              </Dollars>
                            </li>
                            <li
                              key="1"
                            >
                              <strong>
                                FFY 
                                1
                                 cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $100
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $100
                                </span>
                              </Dollars>
                            </li>
                            <li
                              key="2"
                            >
                              <strong>
                                FFY 
                                2
                                 cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $200
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $200
                                </span>
                              </Dollars>
                            </li>
                          </ul>
                        </div>
                      </div>
                      <div
                        className="nowrap visibility--screen"
                      >
                        <Button
                          onClick={[Function]}
                          size="small"
                          type="button"
                          variation="transparent"
                        >
                          <button
                            className="ds-c-button ds-c-button--transparent ds-c-button--small"
                            onClick={[Function]}
                            type="button"
                          >
                            Edit
                          </button>
                        </Button>
                        |
                        <Button
                          onClick={[Function]}
                          size="small"
                          type="button"
                          variation="transparent"
                        >
                          <button
                            className="ds-c-button ds-c-button--transparent ds-c-button--small"
                            onClick={[Function]}
                            type="button"
                          >
                            Remove
                          </button>
                        </Button>
                      </div>
                    </div>
                  </Review>
                </StandardReview>
              </div>
              <div
                className="visibility--print"
              >
                <StandardReview
                  editHref={null}
                  heading="1. person name"
                  onDeleteClick={null}
                  onEditClick={null}
                >
                  <Review
                    editContent={
                      <div
                        className="nowrap visibility--screen"
                      />
                    }
                    editText="Edit"
                    heading="1. person name"
                  >
                    <div
                      className="ds-c-review ds-u-border-bottom--2 ds-u-padding-y--2 ds-u-justify-content--between ds-u-display--flex"
                    >
                      <div
                        className="ds-u-margin-right--2"
                      >
                        <h3
                          className="ds-c-review__heading ds-text ds-u-margin-bottom--0 ds-u-font-weight--bold ds-u-display--inline-block"
                        >
                          1. person name
                        </h3>
                        <div
                          className="ds-c-review__body"
                        >
                          <ul
                            className="ds-c-list--bare"
                          >
                            <li>
                              Primary APD Point of Contact
                            </li>
                            <li>
                              unobservable
                            </li>
                            <li>
                              person1@theplace.gov
                            </li>
                            <li>
                              <strong>
                                Time commitment to project:
                              </strong>
                               
                              27
                              %
                            </li>
                            <li>
                              <strong>
                                Total cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $300
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $300
                                </span>
                              </Dollars>
                            </li>
                            <li
                              key="1"
                            >
                              <strong>
                                FFY 
                                1
                                 cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $100
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $100
                                </span>
                              </Dollars>
                            </li>
                            <li
                              key="2"
                            >
                              <strong>
                                FFY 
                                2
                                 cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $200
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $200
                                </span>
                              </Dollars>
                            </li>
                          </ul>
                        </div>
                      </div>
                      <div
                        className="nowrap visibility--screen"
                      />
                    </div>
                  </Review>
                </StandardReview>
              </div>
            </ApdStateKeyPerson>
          </div>
        </FormAndReviewItem>
        <FormAndReviewItem
          collapsedComponent={[Function]}
          expandedComponent={[Function]}
          index={1}
          initialCollapsed={true}
          item={
            Object {
              "costs": Object {
                "1": 0,
                "2": 0,
              },
              "email": "person2@theplace.gov",
              "hasCosts": false,
              "key": "person2",
              "name": "",
              "percentTime": 72,
              "position": "",
            }
          }
          key="person2"
          onDeleteClick={[Function]}
          years={
            Array [
              "1",
              "2",
            ]
          }
        >
          <div
            className="form-and-review-list--item__collapsed"
          >
            <ApdStateKeyPerson
              expand={[Function]}
              index={1}
              item={
                Object {
                  "costs": Object {
                    "1": 0,
                    "2": 0,
                  },
                  "email": "person2@theplace.gov",
                  "hasCosts": false,
                  "key": "person2",
                  "name": "",
                  "percentTime": 72,
                  "position": "",
                }
              }
              onDeleteClick={[Function]}
              years={
                Array [
                  "1",
                  "2",
                ]
              }
            >
              <div
                className="visibility--screen"
              >
                <StandardReview
                  editHref={null}
                  heading="2. "
                  headingLevel={4}
                  onDeleteClick={[Function]}
                  onEditClick={[Function]}
                >
                  <Review
                    editContent={
                      <div
                        className="nowrap visibility--screen"
                      >
                        <Button
                          onClick={[Function]}
                          size="small"
                          type="button"
                          variation="transparent"
                        >
                          Edit
                        </Button>
                        <React.Fragment>
                          |
                          <Button
                            onClick={[Function]}
                            size="small"
                            type="button"
                            variation="transparent"
                          >
                            Remove
                          </Button>
                        </React.Fragment>
                      </div>
                    }
                    editText="Edit"
                    heading="2. "
                    headingLevel={4}
                  >
                    <div
                      className="ds-c-review ds-u-border-bottom--2 ds-u-padding-y--2 ds-u-justify-content--between ds-u-display--flex"
                    >
                      <div
                        className="ds-u-margin-right--2"
                      >
                        <h4
                          className="ds-c-review__heading ds-text ds-u-margin-bottom--0 ds-u-font-weight--bold ds-u-display--inline-block"
                        >
                          2. 
                        </h4>
                        <div
                          className="ds-c-review__body"
                        >
                          <ul
                            className="ds-c-list--bare"
                          >
                            <li />
                            <li>
                              <strong>
                                Total cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $0
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $0
                                </span>
                              </Dollars>
                            </li>
                          </ul>
                        </div>
                      </div>
                      <div
                        className="nowrap visibility--screen"
                      >
                        <Button
                          onClick={[Function]}
                          size="small"
                          type="button"
                          variation="transparent"
                        >
                          <button
                            className="ds-c-button ds-c-button--transparent ds-c-button--small"
                            onClick={[Function]}
                            type="button"
                          >
                            Edit
                          </button>
                        </Button>
                        |
                        <Button
                          onClick={[Function]}
                          size="small"
                          type="button"
                          variation="transparent"
                        >
                          <button
                            className="ds-c-button ds-c-button--transparent ds-c-button--small"
                            onClick={[Function]}
                            type="button"
                          >
                            Remove
                          </button>
                        </Button>
                      </div>
                    </div>
                  </Review>
                </StandardReview>
              </div>
              <div
                className="visibility--print"
              >
                <StandardReview
                  editHref={null}
                  heading="2. "
                  onDeleteClick={null}
                  onEditClick={null}
                >
                  <Review
                    editContent={
                      <div
                        className="nowrap visibility--screen"
                      />
                    }
                    editText="Edit"
                    heading="2. "
                  >
                    <div
                      className="ds-c-review ds-u-border-bottom--2 ds-u-padding-y--2 ds-u-justify-content--between ds-u-display--flex"
                    >
                      <div
                        className="ds-u-margin-right--2"
                      >
                        <h3
                          className="ds-c-review__heading ds-text ds-u-margin-bottom--0 ds-u-font-weight--bold ds-u-display--inline-block"
                        >
                          2. 
                        </h3>
                        <div
                          className="ds-c-review__body"
                        >
                          <ul
                            className="ds-c-list--bare"
                          >
                            <li />
                            <li>
                              person2@theplace.gov
                            </li>
                            <li>
                              <strong>
                                Time commitment to project:
                              </strong>
                               
                              72
                              %
                            </li>
                            <li>
                              <strong>
                                Total cost:
                              </strong>
                               
                              <Dollars
                                long={false}
                              >
                                <span
                                  className="visibility--screen"
                                >
                                  $0
                                </span>
                                <span
                                  className="visibility--print"
                                >
                                  $0
                                </span>
                              </Dollars>
                            </li>
                          </ul>
                        </div>
                      </div>
                      <div
                        className="nowrap visibility--screen"
                      />
                    </div>
                  </Review>
                </StandardReview>
              </div>
            </ApdStateKeyPerson>
          </div>
        </FormAndReviewItem>
        <Button
          className="visibility--screen"
          onClick={[Function]}
          type="button"
        >
          <button
            className="ds-c-button visibility--screen"
            onClick={[Function]}
            type="button"
          >
            Add another person
          </button>
        </Button>
      </div>
    </FormAndReviewList>
  </ApdStateKeyPersonnel>
</Provider>
`;
